core/wave-autoscale/src/scaling_component/k8s_patch.rs

```diff
@@ -0,0 +1,227 @@
+use super::ScalingComponent;
+use async_trait::async_trait;
+use data_layer::ScalingComponentDefinition;
+use kube::{
+    api::{Api, DynamicObject, Patch, PatchParams},
+    discovery, Client,
+};
+use std::collections::HashMap;
+
+pub struct K8sPatchScalingComponent {
+    definition: ScalingComponentDefinition,
+}
+
+impl K8sPatchScalingComponent {
+    pub const SCALING_KIND: &'static str = "kubernetes-patch";
+
+    pub fn new(definition: ScalingComponentDefinition) -> Self {
+        K8sPatchScalingComponent { definition }
+    }
+
+    async fn get_client(
+        &self,
+        _api_server_endpoint: Option<String>,
+        _ca_cert: Option<String>,
+        _namespace: Option<String>,
+    ) -> anyhow::Result<kube::Client> {
+        // TODO: Use the metadata to create a Kubernetes Client
+        // Infer the runtime environment and try to create a Kubernetes Client
+        let client = Client::try_default().await?;
+        Ok(client)
+    }
+}
+
+#[async_trait]
+impl ScalingComponent for K8sPatchScalingComponent {
+    fn get_scaling_component_kind(&self) -> &str {
+        &self.definition.component_kind
+    }
+    fn get_id(&self) -> &str {
+        &self.definition.id
+    }
+
+    async fn apply(&self, params: HashMap<String, serde_json::Value>) -> anyhow::Result<()> {
+        let metadata = self.definition.metadata.clone();
+
+        let (
+            Some(serde_json::Value::String(namespace)),
+            Some(serde_json::Value::String(name)),
+            Some(serde_json::Value::Object(manifest)) 
+            ) = (
+            params.get("namespace"),
+            params.get("name"),
+            params.get("manifest"),
+        ) else {
+            return Err(anyhow::anyhow!("Invalid metadata"));
+        };
+
+        // TODO: Use the metadata to create a Kubernetes Client
+        let api_server_endpoint = metadata
+                .get("api_server_endpoint")
+                .map(|api_server_endpoint| api_server_endpoint.to_string());
+        let ca_cert = metadata
+                .get("ca_cert")
+                .map(|ca_cert| ca_cert.to_string());
+        let Ok(client) = self.get_client(api_server_endpoint, ca_cert, Some(namespace.to_string())).await else {
+            return Err(anyhow::anyhow!("cannot create kubernetes client"));
+        };
+
+        println!("manifest:\n {:#?}", manifest);
+        let api_version = manifest.get("apiVersion").unwrap();
+        let api_group = api_version.as_str().unwrap().split("/").nth(0).unwrap();
+        let kind = manifest.get("kind").unwrap().as_str().unwrap();
+
+        let apigroup = discovery::group(&client, api_group).await.unwrap();
+        let (ar, caps) = apigroup.recommended_kind(kind).unwrap();
+        let api: Api<DynamicObject> = Api::namespaced_with(client, namespace, &ar);
+
+        let patch_params = PatchParams::apply("wave-autoscale");
+        let patch_params = PatchParams::force(patch_params);
```

Comment:
Add a comment why use apply and force together

Reply:
There was a conflict with terraform, so added the force option.
`"Apply failed with 1 conflict: conflict with \"HashiCorp\" using apps/v1: .spec.replicas"`